lib/internal/http2/core.js

```diff
@@ -1479,7 +1479,15 @@ class ClientHttp2Session extends Http2Session {
 
     const onConnect = requestOnConnect.bind(stream, headersList, options);
     if (this.connecting) {
-      this.once('connect', onConnect);
+      if (this._pendingRequestCalls) {
+        this._pendingRequestCalls.push(onConnect);
```

Comment:
Can we instead hide these behind a symbol instead of an underscore-prefixed property?

Reply:
Good idea, I'll do that.